api/common/upgradeseries.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package common
+
+import (
+	"github.com/juju/errors"
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/juju/api/base"
+	apiwatcher "github.com/juju/juju/api/watcher"
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/watcher"
+)
+
+type UpgradeSeriesAPI struct {
+	facade base.FacadeCaller
+	tag    names.Tag
+}
+
+func NewUpgradeSeriesAPI(facade base.FacadeCaller, tag names.Tag) *UpgradeSeriesAPI {
+	return &UpgradeSeriesAPI{facade: facade, tag: tag}
+}
+
+// WatchActionNotifications returns a StringsWatcher for observing the state of
+// a series upgrade.
+func (u *UpgradeSeriesAPI) WatchUpgradeSeriesNotifications() (watcher.NotifyWatcher, error) {
+	var results params.NotifyWatchResults
+	args := params.Entities{
+		Entities: []params.Entity{{Tag: u.tag.String()}},
+	}
+	err := u.facade.FacadeCall("WatchUpgradeSeriesNotifications", args, &results)
+	if err != nil {
+		return nil, errors.Annotatef(err, "from api/common/upgradeseries FacadeCall(%s)", u.facade.Name())
+	}
+	if len(results.Results) != 1 {
+		return nil, errors.Errorf("expected 1 result, got %d", len(results.Results))
+	}
+	result := results.Results[0]
+	if result.Error != nil {
+		return nil, result.Error
+	}
+	w := apiwatcher.NewNotifyWatcher(u.facade.RawAPICaller(), result)
+	return w, nil
+}
+
+// UpgradeSeriesStatus returns the upgrade series status of a unit from remote state
+func (u *UpgradeSeriesAPI) UpgradeSeriesStatus() (string, error) {
+	var results params.UpgradeSeriesStatusResults
+	args := params.Entities{
+		Entities: []params.Entity{{Tag: u.tag.String()}},
+	}
+	err := u.facade.FacadeCall("UpgradeSeriesStatus", args, &results)
+	if err != nil {
+		return "", err
+	}
+	if len(results.Results) != 1 {
+		return "", errors.Errorf("expected 1 result, got %d", len(results.Results))
+	}
+	result := results.Results[0]
+	if result.Error != nil {
+		//TODO (externalreality) The code to do convert api errors (with
+		//error codes) back to normal Go errors is in bad spot and
+		//causes import cycles which is why we don't use it here and may
+		//be the reason why it has few uses despite being useful.
+		if params.IsCodeNotFound(result.Error) {
+			return "", errors.NewNotFound(result.Error, "")
+		}
+		return "", result.Error
+	}
+	return result.Status, nil
+}
+
+// UpgradeSeriesStatus sets the upgrade series status of the unit in the remote state
+func (u *UpgradeSeriesAPI) SetUpgradeSeriesStatus(status string) error {
+	var results params.ErrorResults
+	args := params.SetUpgradeSeriesStatusParams{
+		[]params.SetUpgradeSeriesStatusParam{{
+			Entity: params.Entity{Tag: u.tag.String()},
+			Status: status,
+		}},
+	}
+	err := u.facade.FacadeCall("SetUpgradeSeriesStatus", args, &results)
+	if err != nil {
+		return err
```

Comment:
This error message is not consistent with the `WatchUpgradeSeriesNotifications()` facade call error message found above.